.github/workflows/ci.yml

```diff
@@ -25,27 +25,24 @@ jobs:
         rust:
           - stable
           # MSRV
-          - 1.56.0
+          - 1.60.0
     steps:
       - uses: actions/checkout@v3
         with:
           persist-credentials: false
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@master
         with:
-          profile: minimal
           toolchain: ${{ matrix.rust }}
           components: clippy
 
       - name: Clippy (${{ matrix.os }})
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy-ci
+        run: cargo clippy-ci
 
       - name: Clippy (Android)
         if: matrix.os == 'ubuntu-latest'
         run: |
           rustup target add aarch64-linux-android
-          cargo install cargo-ndk
+          cargo install cargo-ndk --version 2.12.7
```

Comment:
> are you suggesting we use the nightly toolchain to build cargo ndk,

Not quite, I was suggesting we use the latest stable instead of our MSRV to build `cargo-ndk` in CI.

> but invoke cargo ndk -t arm64-v8a clippy-ci and ./gradlew connectedDebugAndroidTest from the MSRV toolchain

That part is correct, yes. Summarized, it'd be like this:

- Building test tools: Latest stable
- Building our own code: MSRV stable

Reply:
Thanks for clarifying :+1: 

> If it's the former, I'll have to do more investigation into whether it's possible to do this with the dtonlay actions. I thought that the installing cargo-ndk with nightly would be isolated from a stable toolchain installed afterwards.

I did some experimenting, and it seems like this model is workable, except `cargo-ndk` doesn't want to support the MSRV toolchain even once installed... :sweat: 

```
Run rustup target add aarch64-linux-android
info: downloading component 'rust-std' for 'aarch64-linux-android'
info: installing component 'rust-std' for 'aarch64-linux-android'
error: Rust compiler is too old and not supported by cargo-ndk.
note: Upgrade Rust to at least v1.68.0.
```

Reply:
I also tried shifting the target installation to happen on stable and only keeping the `cargo ndk` invocation on MSRV and it failed with the same error when running `cargo ndk -t arm64-v8a clippy-msrv-ci `

Reply:
That is frustrating. Apparently they have a [hard-check for their MSRV](https://github.com/bbqsrc/cargo-ndk/blob/de33c32a16e74ecf2f19f7fc624099ecf89632a1/src/cli.rs#L288-L292), which I hadn't originally expected. I wonder if the [Android NDK updates](https://blog.rust-lang.org/2023/03/09/Rust-1.68.0.html#other-changes) in that version caused them enough trouble to cut support for below.

I'm not quite sure what to do here that doesn't involve using a higher version for Android, especially since `memchr` has made their new MSRV 1.62. It'd be nice to not need `cargo ndk` for the check CI job, but _ring_ requires it.

Reply:
>  especially since memchr has made their new MSRV 1.62.

Just to feel things out I tried bumping MSRV to 1.62 and explicitly using `cargo ndk` 2.12.7 and it bumped into another dep requiring a higher MSRV:
> package `malloced v1.3.1` cannot be built because it requires rustc 1.64 or newer, while the currently active rustc version is 1.62.0

Putting the MSRV at 1.64 and explicitly using `cargo ndk` 2.12.7 worked OK. Is that potentially an option? 1.64 is almost a year old.

> I'm not quite sure what to do here that doesn't involve using a higher version for Android, especially since memchr has made their new MSRV 1.62. It'd be nice to not need cargo ndk for the check CI job, but ring requires it.

In the near-ish future I think we could swap ring out, but I think it will be the default backend for some time so having test coverage with it is probably sensible.

Reply:
I think 1.64 is fine (assuming it works for 1P).

Reply:
> In the near-ish future I think we could swap ring out, but I think it will be the default backend for some time so having test coverage with it is probably sensible.

FWIW when other providers are available we should support them here, but I'd want to test this crate with both instead of just avoiding _ring_ outright.

> I think 1.64 is fine (assuming it works for 1P).

Agreed, and it shouldn't be a problem for 1P. We just updated to 1.72 today.

Reply:
I updated this branch based on a MSRV of 1.64.